### PR TITLE
feat: assorted new views, lots of QA/fixes and adherence to design of past pages

### PR DESCRIPTION
**Pull request by madhuravius**
- [x] Otp is improved slightly but we need a better warning message (it bubbles up from middleware right now)

Misc

- [x] disabled opacity and cursor disabled
- [x] collapsible how it works at the bottom of sign up
- [x] labels on billing should be same font size (no need to make it smaller), so make the credit card one row and break up the date/confirmation
- [x] first letter on all buttons needs to be caps
- [x] `font-bold` on all the h1 types
- [x] on 404 page - capitalize the word page (`View Status Page`)

Lower priority or pt2
- [ ] Billing API calls (trials checks, billing check, email verification check)
- [ ] Plan selection wrap up
- [ ] Actual overall flow (with trials disabled)
- [ ] More meaningful unit tests from backend errors etc (not just flat out renders)

---

Contains work for:

* https://app.shortcut.com/aptible/story/13216/update-existing-pages-with-proper-designs (assorted fixes)
* https://app.shortcut.com/aptible/story/13219/implement-check-email-page (already there, mostly reskinned)
* https://app.shortcut.com/aptible/story/13955/implement-add-billing-method-page (missing actual rest calls, non-trivial)

To be pushed:

* https://app.shortcut.com/aptible/story/13954/implement-plan-selection

---

See screenshots in this thread: https://github.com/aptible/app-ui/pull/117#issuecomment-1554499736

### CODE REVIEW

File: src/ui/layouts/unauth-required.test.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      initEntries: ["/mock"],
+      additionalRoutes: [
+        {
+          path: homeUrl(),
+          element: <p>Simulated home</p>,
+        },
+      ],
+    });
+    server.use(
+      ...[
```
Comment (neurosnap) [inline src/ui/layouts/unauth-required.test.tsx:22]: You can just reference the single call, no need to spread it from an array

File: src/ui/layouts/unauth-required.test.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      <TestProvider>
+        <div>
+          <UnauthRequired />
+          <p>Test element</p>
+        </div>
+      </TestProvider>,
+    );
+    const el = await screen.findByText("Test element");
+    const home = await screen.queryByText("Simulated home");
+    await waitFor(() => {
```
Comment (neurosnap) [inline src/ui/layouts/unauth-required.test.tsx:38]: I'm curious, why did you need this?

File: src/ui/layouts/unauth-required.test.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      <TestProvider>
+        <div>
+          <UnauthRequired />
+          <p>Test element</p>
+        </div>
+      </TestProvider>,
+    );
+    const el = await screen.queryByText("Test element");
+    const home = await screen.findByText("Simulated home");
+    await waitFor(() => {
```
Comment (neurosnap) [inline src/ui/layouts/unauth-required.test.tsx:65]: Same here

File: src/ui/pages/billing-method.test.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+describe("Billing Method page", () => {
+  it("the billing method page should render", async () => {
+    const { TestProvider } = setupIntegrationTest();
+    render(
+      <TestProvider>
+        <BillingMethodPage />
+      </TestProvider>,
+    );
+    const el = await screen.findByRole("button");
+    await waitFor(() => {
```
Comment (neurosnap) [inline src/ui/pages/billing-method.test.tsx:14]: Same here

File: src/ui/pages/signup.tsx [hunk 5]
```diff
@@ -124,106 +122,129 @@ const SignupPageForm = () => {
     }
   });
 
+  // presentError - this value is set because in specific scenarios, we do not need the
+  // middleware error message presented - namely on Unauthorized checks, as we do a
+  // validation of the current token to see if a user is loaded. for all other
+  const presentError = loader.isError && loader.message !== "Unauthorized";
```
Comment (neurosnap) [inline src/ui/pages/signup.tsx:128]: Do you still need this?

File: src/ui/pages/verify-email.test.tsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  });
+  it("the verify email page should properly fail", async () => {
+    const { TestProvider } = setupIntegrationTest();
+    render(
+      <TestProvider>
+        <VerifyEmailPage />
+      </TestProvider>,
+    );
+    server.use(
+      ...[
```
Comment (neurosnap) [inline src/ui/pages/verify-email.test.tsx:29]: Don't need this.

File: src/ui/pages/verify-email.test.tsx [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+        rest.get(`${testEnv.authUrl}/verifications`, (_, res, ctx) => {
+          return res(ctx.status(401));
+        }),
+      ],
+    );
+    const el = await screen.findByRole("button");
+    const errorText = await screen.queryByText("Failed to verify your email");
+    await waitFor(() => {
```
Comment (neurosnap) [inline src/ui/pages/verify-email.test.tsx:37]: Same here.

File: src/ui/layouts/unauth-required.test.tsx (hunk 2 above)
Comment (madhuravius) [reply src/ui/layouts/unauth-required.test.tsx:38]: i can see again but navigate didn't work immediately, i can try retesting

i ended up falling back to this due to async issues (tests hanging)

File: src/ui/layouts/unauth-required.test.tsx (hunk 1 above)
Comment (madhuravius) [reply src/ui/layouts/unauth-required.test.tsx:22]: i tried this and it got mad, will try again

File: src/ui/pages/signup.tsx (hunk 5 above)
Comment (madhuravius) [reply src/ui/pages/signup.tsx:128]: yeah this is on signup, it will always render an error if this is not done (since no current token is available)

File: src/ui/pages/verify-email.test.tsx (hunk 6 above)
Comment (madhuravius) [reply src/ui/pages/verify-email.test.tsx:29]: oh the shallow copy? yeah ill try dropping

File: src/deploy/plan/index.ts
Comment (neurosnap) [inline src/deploy/plan/index.ts:1]: Do we still need this?

File: src/ui/pages/create-project-git-settings.test.tsx
Comment (neurosnap) [inline src/ui/pages/create-project-git-settings.test.tsx:1]: This is weird, what happened here?

File: src/ui/pages/signup.test.tsx [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+describe("Signup page", () => {
+  it("the sign up page should render", async () => {
+    const { TestProvider } = setupIntegrationTest();
+    render(
+      <TestProvider>
+        <SignupPage />
+      </TestProvider>,
+    );
+    const el = await screen.findByRole("button");
+    await waitFor(() => {
```
[the remark this answers is not in the document]
Comment (madhuravius) [reply src/ui/pages/signup.test.tsx:14]: this should've been removed, looking again

[the remark this answers is not in the document]
Comment (madhuravius) [reply src/ui/pages/signup.test.tsx:14]: this should not have been removed, this is a good use case for why it existed 

image

[the remark this answers is not in the document]
Comment (madhuravius) [reply src/ui/pages/signup.test.tsx:14]: it's possible i added this to other pages for the same reason (eventually settled type behavior) but i undid it as a result

[the remark this answers is not in the document]
Comment (madhuravius) [reply src/ui/pages/signup.test.tsx]: better way to handle this is via more granular await

File: src/ui/pages/plans.tsx
[the remark this answers is not in the document]
Comment (madhuravius) [reply src/ui/pages/plans.tsx:1]: for pt2

**Review by neurosnap — APPROVED** [review]
